rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java

```diff
@@ -182,23 +183,23 @@ void sample(RrdDb dataBase, long timeInMs) throws IOException {
                 }
             } catch (NumberFormatException ex) {
                 // do not save non-numeric values
-                if (logger.isTraceEnabled()) {
+                String exceptionAsString = StackTraceUtil.getStackTrace(ex).replaceAll("For input string: \".*\"", "");
+                exceptionAsString = dataSource + " / " + fullName + exceptionAsString;
+
+                if (!exceptionsThrown.contains(exceptionAsString)) {
+                    exceptionsThrown.add(exceptionAsString);
+                    logger.warn("Non numeric value for " + dataSource + " / " + fullName + ": " + ex.getMessage());
+                } else {
                     logger.trace("Non numeric value for " + dataSource + " / " + fullName + ": " + ex.getMessage());
                 }
-            } catch (InstanceNotFoundException | AttributeNotFoundException e) {
-                if (logger.isTraceEnabled()) {
-                    logger.trace("Cannot read attribute " + attrName + " for object " + objectName + ": " +
-                                 e.getMessage());
-                }
-            } catch (UnsupportedOperationException e) {
-                if (logger.isTraceEnabled()) {
+            } catch (Exception e) {
+                String exceptionAsString = StackTraceUtil.getStackTrace(e);
+                if (!exceptionsThrown.contains(exceptionAsString)) {
+                    exceptionsThrown.add(exceptionAsString);
                     logger.warn("Error while reading attribute " + attrName + " for object " + objectName + ": ", e);
```

Comment:
i dont understand your point, if the log's war level is not enabled, the string will be built anyway, otherwise this string build step will be skipped, why this will not bring anything in terms of performance?